src/Reflection/ReflectionFunctionAbstract.php

```diff
@@ -270,17 +273,34 @@ public function isClosure() : bool
         return $this->node instanceof Node\Expr\Closure;
     }
 
-    /**
-     * Is this function deprecated?
-     *
-     * Note - we cannot reflect on internal functions (as there is no PHP source
-     * code we can access. This means, at present, we can only EVER return false
-     * from this function.
-     *
-     * @see https://github.com/Roave/BetterReflection/issues/38
-     */
     public function isDeprecated() : bool
     {
+        $docComment = $this->getDocComment();
+        if ($docComment !== '') {
+            $docBlockFactory = DocBlockFactory::createInstance();
+            $docBlock = $docBlockFactory->create($docComment);
+            if ($docBlock->hasTag('deprecated')) {
+                return true;
+            }
+        }
+
+        foreach ($this->getAttributes() as $attribute) {
+            if ($attribute->getName() === 'JetBrains\PhpStorm\Deprecated') {
+                $attributeArgs = $attribute->getArguments();
+                if (isset($attributeArgs['since'])) {
+                    // Convert version string to PHP_VERSION_ID equivalent.
+                    [$phpMajor, $phpMinor] = explode('.', $attributeArgs['since']);
+                    $phpMajor = (int) str_pad($phpMajor, 2, '0');
+                    $phpMinor = (int) str_pad($phpMinor, 2, '0');
+                    $phpPatch = 00;
+                    $phpVersionId = (int) ($phpMajor.$phpMinor.$phpPatch);
```

Comment:
maybe

```suggestion
                    $phpVersionId = sprintf('%02d%02d00', $phpMajor, $phpMinor)
```

see https://3v4l.org/5EpvM

Reply:
Thanks!